src/server/binary/binaryServerStarter.ts

```diff
@@ -64,9 +64,19 @@ async function getServerBinaryPath(): Promise<string> {
   const binaryPath: string = config.get("server.binary.path");
   if (binaryPath) {
     if (fs.existsSync(binaryPath)) {
-      return Promise.resolve(binaryPath);
+      try {
+        // checks execute permission
+        fs.accessSync(binaryPath, fs.constants.X_OK)
+        // checks read/write permission
+        fs.accessSync(binaryPath, fs.constants.R_OK | fs.constants.W_OK)
```

Comment:
Can the 2 lines be combined into one ? Also we don't need write permissions for this to work so I would just remove it.

```suggestion
        // checks read/execute permission
        fs.accessSync(binaryPath, fs.constants.R_OK | fs.constants.X_OK)
```

Reply:
I would also remove the `console.error(..)`.

Reply:
Do you think there's any value in have 2 separate messages for permissions (i.e. one for executable permissions and one for read? Otherwise I can combine both, and leave the generic permission denied message.